assets/environment-interpreter/wrapper/wrapper

```diff
@@ -117,6 +140,26 @@ if [ -z "${FLOX_ENV-}" ]; then
   exit 1
 fi
 
+# If invoked with the --keep-env-vars option, then clear the environment of
+# all variables _not_ found in this list.
+# shellcheck disable=SC2034
+if [ -n "${_flox_keep_env_vars:-}" ]; then
+  while IFS='=' read -r -d '' n v; do
+    # Keep env vars starting with "_".
+    if [[ "$n" == _* ]]; then
+      wrapperDebug "Keeping env var: $n=$v"
+    else
+      # Discard env vars not mentioned in the list.
+      case " FLOX_ENV ${_flox_keep_env_vars:-} " in
+      *" $n "*) wrapperDebug "Keeping env var: $n=$v";;
+      *) wrapperDebug "Discarding env var: $n=$v";
+         unset "$n"
+         ;;
+      esac
+    fi
+  done < <("$_env" -0)
+fi
+
```

Comment:
blocking: I don't think this logic belongs in the wrapper. Can you unset any variables you need to unset at the start of the build script?

Reply:
I'm not sure if my comment was very clear or not. What I was suggesting is currently there's `$(_bash) -e $($(_pvarname)_buildScript)`. Could the start of that buildscript be something like what you have here? Or is that what you're saying is error-prone?
```
  while IFS='=' read -r -d '' n v; do
    # Keep env vars starting with "_".
    if [[ "$n" == _* ]]; then
      wrapperDebug "Keeping env var: $n=$v"
    else
      # Discard env vars not mentioned in the list.
      case " FLOX_ENV ${_flox_keep_env_vars:-} " in
      *" $n "*) wrapperDebug "Keeping env var: $n=$v";;
      *) wrapperDebug "Discarding env var: $n=$v";
         unset "$n"
         ;;
      esac
    fi
  done < <("$_env" -0)
```

Reply:
We discussed creating the equivalent of `env (-k|--keep) <var>` which we both agree should have been part of `env` in the first place! We can write that and sandwich it in between the outer and inner activations, similar to what we were doing before. One of us will get that done asap, based on what time we have available.

Reply:
Created new `env-filter` script in commit 4425712a1937764c2eea95f85d59ba96fcb6848c and updated `flox-build.mk` to use it between the outer and inner activations. It's very much a DOTADIW script. 😄